src/plugins/event-plugins/NavigationPlugin.ts

```diff
@@ -184,11 +199,24 @@ export class NavigationPlugin extends InternalPlugin {
     /**
      * W3C specification: https://www.w3.org/TR/navigation-timing-2/#bib-navigation-timing
      */
-    performanceNavigationEventHandlerTimingLevel2 = (entryData: any): void => {
+    performanceNavigationEventHandlerTimingLevel2 = (
+        entryData: PerformanceNavigationTiming
+    ): void => {
+        if (this.config.ignore(entryData)) {
+            return;
+        }
+
         const eventDataNavigationTimingLevel2: NavigationEvent = {
             version: '1.0.0',
-            initiatorType: entryData.initiatorType,
-            navigationType: entryData.type,
+            initiatorType: entryData.initiatorType as
+                | 'navigation'
+                | 'route_change',
```

Comment:
question: where does the `route_change` value come from? I don't see it on the MDN docs for `PerformanceResourceTiming.initatorType`

https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming/initiatorType

Reply:
I copied and pasted the type that TypeScript resolved. The NavigationEvent type is created from [the corresponding JSON schema](https://github.com/aws-observability/aws-rum-web/blob/main/src/event-schemas/navigation-event.json#L12-L20).

You are correct in that it would be more accurate to omit `route_change` here, since it will never appear in the PerformanceNavigationTiming object provided by the browser.

I assume the same is true of `entryData.type`.

Reply:
Oh I see. Looks like something we can address in a separate change.